fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -899,12 +904,14 @@ struct EncryptedRangeFileWriter : public IRangeFileWriter {
 		if (k.size() > 0 &&
 		    (!self->cipherKeys.headerCipherKey.present() || !self->cipherKeys.headerCipherKey.get().isValid() ||
 		     !self->cipherKeys.textCipherKey.isValid())) {
-			wait(updateEncryptionKeysCtx(self, k));
+			wait(updateEncryptionKeysCtx(self, k, true));
 		}
 		// Need to account for extra "empty" value being written in the case of crossing tenant boundaries
 		int toWrite = sizeof(uint32_t) + k.size() + sizeof(uint32_t);
 		wait(newBlockIfNeeded(self, toWrite));
-		bool createdNewBlock = wait(finishCurTenantBlockStartNewIfNeeded(self, k, StringRef(), false));
+		// We want to check the tenant cache here since the first/last key for a block may not be a valid KV pair (in
```

Comment:
My understanding is `first` and/or `last` Key would still be a valid TenantPrefix, not sure why would it use `default domain`?

Reply:
The very first and last key in a file (not a block) may not always be a key in the database (hence it may not be associated with a tenant). In these case attempting to get a tenant prefix will result in an invalid tenant id. This is the case because in a backup file we can represent empty ranges this way (i.e if there are no KV pairs in the database we still want the snapshot to contain a begin and end key)